tests/e2e/vsphere_volume_expansion.go

```diff
@@ -1796,13 +1860,27 @@ func invokeTestForExpandVolumeMultipleTimes(f *framework.Framework, client clien
 	pod, err := createPod(client, namespace, nil, []*v1.PersistentVolumeClaim{pvclaim}, false, execCommand)
 	gomega.Expect(err).NotTo(gomega.HaveOccurred())
 
+	defer func() {
+		ginkgo.By("Deleting the pod")
+		err = fpod.DeletePodWithWait(client, pod)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+	}()
+
 	var vmUUID string
-	ginkgo.By(fmt.Sprintf("Verify volume: %s is attached to the node: %s", pv.Spec.CSI.VolumeHandle, pod.Spec.NodeName))
-	vmUUID = getNodeUUID(client, pod.Spec.NodeName)
-	if guestCluster {
+	var exists bool
+	ginkgo.By(fmt.Sprintf("Verify volume: %s is attached to the node: %s", volHandle, pod.Spec.NodeName))
+	if vanillaCluster {
+		vmUUID = getNodeUUID(client, pod.Spec.NodeName)
+	} else if guestCluster {
 		vmUUID, err = getVMUUIDFromNodeName(pod.Spec.NodeName)
 		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+	} else {
+		annotations := pod.Annotations
+		vmUUID, exists = annotations[vmUUIDLabel]
+		gomega.Expect(exists).To(gomega.BeTrue(), fmt.Sprintf("Pod doesn't have %s annotation", vmUUIDLabel))
+		framework.Logf("VMUUID : %s", vmUUID)
```

Comment:
Same as the comment above from Ashrith, log `VMUUID` outside `if.. else`

Reply:
done